client/src/components/formHelpers/form.tsx

```diff
@@ -35,43 +36,58 @@ function validateFormValues(
   formValues: FormValues,
   options: FormOptions
 ): ValidatedValues {
-  const { isEditorLinkAllowed, isLocalLinkAllowed, types } = options;
+  const {
+    isEditorLinkAllowed,
+    isLocalLinkAllowed,
+    isSourceCodeLinkRequired,
+    types
+  } = options;
   const validatedValues: ValidatedValues = {
     values: {},
     errors: [],
     invalidValues: []
   };
-  const urlValues = Object.entries(formValues).reduce(
-    (result, [key, value]) => {
-      // NOTE: pathValidator is not used here, because it is only used as a
-      // suggestion - should not prevent form submission
-      const validators = [fCCValidator, httpValidator];
-      const isSolutionLink = key !== 'githubLink';
-      if (isSolutionLink && !isEditorLinkAllowed) {
-        validators.push(editorValidator);
-      }
-      if (!isLocalLinkAllowed) {
-        validators.push(localhostValidator);
-      }
 
-      const nullOrWarning = composeValidators(...validators)(value);
-      if (nullOrWarning) {
-        validatedValues.invalidValues.push(nullOrWarning);
-      }
-      if (value && types && types[key] === 'url') {
-        try {
-          value = normalizeUrl(value, normalizeOptions);
-        } catch (err: unknown) {
-          validatedValues.errors.push({
-            error: err as { message?: string },
-            value
-          });
-        }
+  const formFields = Object.entries(formValues);
+
```

Comment:
```suggestion
  // We don't always get a githubLink field in formValues, so we can't simply
  // validate that field like the others. We have to handle it separately.
```

Reply:
I spent way too long trying to come up with a better comment. But, I will just cave.

Reply:
I spent way too long trying to find a simple solution :shrug: